test/language/literals/regexp/u-null-character-escape.js

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2020 Alexey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: prod-CharacterEscape
+description: >
+  Null character escape is permitted in Unicode patterns.
+info: |
+  CharacterEscape[U] ::
+    ControlEscape
+    c ControlLetter
+    0 [lookahead ∉ DecimalDigit]
+    HexEscapeSequence
+    RegExpUnicodeEscapeSequence[?U]
+    IdentityEscape[?U]
+
+  DecimalDigit :: one of
+    0 1 2 3 4 5 6 7 8 9
+---*/
+
+assert(/\0/u.test(String.fromCharCode(0)));
+assert(/\0a/u.test('\0a'));
+assert(/\0②/u.test('\x00②'));
+assert(/\0፬/u.test('\u0000፬'));
```

Comment:
Since the intention seems to be to match the entire string, maybe add `^` and `$`?

Reply:
GraalJS fails the test without `^` and `$`, but passes with them.
Weird, I reverted adding the assertions.

Reply:
Maybe test both then? Or use `match` and check what is being matched?

Reply:
Resolved in 7da7f38, thank you.

It seems like GraalJS parses `\0` differently in beginning of the pattern vs as second char.
//cc @iamstolis

Reply:
Woah, good catch! Interesting bug.

Reply:
> It seems like GraalJS parses \0 differently at the end of the pattern vs not.

Thanks for the report. Yes, this is a bug in GraalJS. We attempt to read a character behind `\0` which is not a good idea when `\0` is the last character of the pattern.